File: Swift/bubbleSort.swift
//The _ means you can call the function with out a labeled arguments like bubbleSort(arr: array)
func bubbleSort(_ arr: [Int]) -> [Int] {
  var sorted = arr
  var right = sorted.count - 1

  while right > 0 {

    for i in 0..<right {
      if sorted[i] > sorted[i + 1] {
        //thou shalt swapeth with thine brethren
        let temp = sorted[i + 1]
        sorted[i + 1] = sorted[i]
        sorted[i] = temp
      }
    }
    right-=1
  }
  return sorted;
}

let array = [5,6,3,4,2,1]
print(bubbleSort(array));

File: C/bubble_sort.c
#include <stdio.h>

void print_array(int arr[], int size);

int main(void)
{
  int arr[] = {8, 4, 43, 2, 1};
  int length = sizeof(arr) / sizeof(arr[0]);
  print_array(arr, length);

  int right = length - 1;
  while (right > 0)
  {
    for (int i = 0; i < right; i++)
    {
      if (arr[i] > arr[i + 1])
      {
        // swap them
        int temp = arr[i];
        arr[i] = arr[i + 1];
        arr[i + 1] = temp;
      }
    }
    right--;
  }

  print_array(arr, length);
}

void print_array(int arr[], int size)
{
  for (int i = 0; i < size; i++)
  {
    printf("%i ", arr[i]);
  }
  printf("\n");
}